auth.go

```diff
@@ -17,6 +17,15 @@ type Auth struct {
 	sync.RWMutex
 }
 
+// NewAuth creates a new default Auth.
+func NewAuth() Auth {
```

Comment:
This was created as the `whitelist`, `banned`, and `ops` maps were nil maps and I couldn't add myself as an Op without it (it panic'd when trying to add to `ops`).

I can do this in a separate PR if you think it's worth the separation.